### PR TITLE
TRAD-309 Coalesce prices in orderbook

### PR DESCRIPTION
**Pull request by DimaCantemir**
TRAD-309 Coalesce prices in orderbook

### CODE REVIEW

**Review by aspin — APPROVED** [review]

File: proto/api.proto [hunk 1]
```diff
@@ -514,6 +520,31 @@ message GetOrderbookResponse {
 message OrderbookItem {
   double price = 1;
   double size = 2;
+  string orderID = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Serum generated OrderID"}];
```
Comment (aspin) [inline proto/api.proto:524]: Should we update descriptions to be `"Serum/Openbook"`?

File: proto/api.proto [hunk 2]
```diff
@@ -363,13 +369,13 @@ service Api {
     };
   }
 
-  rpc GetTickersStream(GetTickersRequest) returns (stream GetTickersStreamResponse) {
+  rpc GetMarketDepthsStream(GetMarketDepthsRequest) returns (stream GetMarketDepthsStreamResponse) {
```
Comment (aspin) [inline proto/api.proto:372]: looks like we're a bit inconsistent with this, but I think we should try to include `StreamRequest` in the parameter name, e.g. `GetMarketDepthsStreamRequest`

File: proto/api.proto [hunk 3]
```diff
@@ -514,6 +520,31 @@ message GetOrderbookResponse {
 message OrderbookItem {
   double price = 1;
   double size = 2;
+  string orderID = 3 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Serum generated OrderID"}];
+  uint64 clientOrderID = 4;
+  string ownerAddress = 5[(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "OpenOrders address for current market"}];
+}
+message GetMarketDepthRequest {
+  string market = 1;
+  uint32 limit = 2 [(grpc.gateway.protoc_gen_openapiv2.options.openapiv2_field) = {description: "Use 0 for no limits"}];
```
Comment (aspin) [inline proto/api.proto:531]: I think we need a `depth` parameter? Are you using limit like that right now?

File: proto/api.proto (hunk 2 above)
Comment (DimaCantemir) [reply proto/api.proto:372]: not really, the requests don't have "stream" in other endpoints.

File: proto/api.proto (hunk 3 above)
Comment (DimaCantemir) [reply proto/api.proto:531]: if you mean the number of price points we return for asks/bids then yes - limit is used for that

**Pull request merged**